applications/GeoMechanicsApplication/custom_elements/interface_element.cpp

```diff
@@ -187,23 +188,32 @@ void InterfaceElement::Initialize(const ProcessInfo& rCurrentProcessInfo)
         GeoElementUtilities::EvaluateShapeFunctionsAtIntegrationPoints(
             mIntegrationScheme->GetIntegrationPoints(), GetGeometry());
 
-    if (!mNeighbourElements.empty()) {
-        for (const auto& element : mNeighbourElements) {
-            std::vector<std::size_t> node_ids_common_with_element(1);
-            std::vector<Vector>      cauchy_stresses;
-            element->CalculateOnIntegrationPoints(CAUCHY_STRESS_VECTOR, cauchy_stresses, rCurrentProcessInfo);
-            const auto nodal_stresses = ExtrapolationUtilities::CalculateNodalVectors(
-                node_ids_common_with_element, element->GetGeometry(),
-                element->GetIntegrationMethod(), cauchy_stresses, element->Id());
-            KRATOS_ERROR_IF_NOT(nodal_stresses.size() == node_ids_common_with_element.size())
-                << " vectors have different sizes" << std::endl;
-        }
-    }
-
     mConstitutiveLaws.clear();
-    for (const auto& r_shape_function_values : shape_function_values_at_integration_points) {
+    for (auto i = std::size_t{0}; i < mIntegrationScheme->GetNumberOfIntegrationPoints(); ++i) {
         mConstitutiveLaws.push_back(GetProperties()[CONSTITUTIVE_LAW]->Clone());
-        mConstitutiveLaws.back()->InitializeMaterial(GetProperties(), GetGeometry(), r_shape_function_values);
+    }
+    // Only interpolate when neighbouring elements that provide nodal stresses were found
+    if (this->Has(NEIGHBOUR_ELEMENTS)) {
```

Comment:
It may be wise to add a second condition here, which checks that the neighbors vector is not empty:
```suggestion
    if (this->Has(NEIGHBOUR_ELEMENTS) && !this->GetValue(NEIGHBOUR_ELEMENTS).empty()) {
```

Reply:
done, although I think that one should not write empty items if they are supposed to contain a list.